src/MSBuildBinLogQuery/Component/BuildComponent.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.Build.Logging.Query.Messaging;
+
+namespace Microsoft.Build.Logging.Query.Component
+{
+    public abstract class BuildComponent
```

Comment:
Should this implement `IQueryableGraphNode`? It feels like you're building up a parallel graph here, can we do it with one set of interfaces/classes?

Reply:
As discussed, I separated node information from nodes themselves. This is because targets have two types of edges and therefore we would like to extract two graphs from targets.